src/basecamp.cpp

```diff
@@ -479,15 +479,34 @@ void basecamp::reset_camp_workers()
     }
 }
 
+void basecamp::add_assignee( character_id id )
+{
+    npc_ptr npc_to_add = overmap_buffer.find_npc( id );
+    if( !npc_to_add ) {
+        debugmsg( "cant find npc to assign to basecamp, on the overmap_buffer" );
+        return;
+    }
+    npc_to_add->assigned_camp = omt_pos;
+    assigned_npcs.push_back( npc_to_add );
+}
+
+void basecamp::remove_assignee( character_id id )
+{
+    npc_ptr npc_to_remove = overmap_buffer.find_npc( id );
+    if( !npc_to_remove ) {
+        debugmsg( "cant find npc to remove from basecamp, on the overmap_buffer" );
+        return;
+    }
+    assigned_npcs.erase( std::remove( assigned_npcs.begin(), assigned_npcs.end(), npc_to_remove ),
+                         assigned_npcs.end() );
+}
+
 void basecamp::validate_assignees()
 {
-    for( auto it2 = assigned_npcs.begin(); it2 != assigned_npcs.end(); ) {
-        auto ptr = *it2;
-        if( ptr->mission != NPC_MISSION_ASSIGNED_CAMP || ptr->global_omt_location() != omt_pos ||
-            ptr->has_companion_mission() ) {
-            it2 = assigned_npcs.erase( it2 );
-        } else {
-            ++it2;
+    for( npc_ptr guy : assigned_npcs ) {
+        if( !guy || !guy->assigned_camp || *guy->assigned_camp != omt_pos ) {
+            assigned_npcs.erase( std::remove( assigned_npcs.begin(), assigned_npcs.end(), guy ),
```

Comment:
You're changing the very same container you're iterating over, that triggers UB. Consider iterating over the container via an iterator and use that iterator to remove the entry.

Reply:
OK now?